### DISCUSSION TITLE
text highlighting

### DISCUSSION THREAD

**Discussion by aetkie** [UNC-Project-Open-AAC/OS-DPI#245] [Ideas]
There's a commercial AAC product that features text highlighting as the speech output reads. How would this be simulated in the OS-DPI? Is it CSS to alter the display/message window?

**Comment by gbishop** [comment]
When I suggested this for Tar Heel Reader many years ago Karen advised that it was a bad idea for literacy. But it might be different for AAC. 

I see there are events available from the Speech Synthesis API that provides the offset into the text that is being spoken. It would require some additional Javascript coding to expose it to you. 

It isn't immediately obvious to me how to declaratively say what you want. Maybe in Actions it looks something like:

|Origin|Conditions|State|New value|
|--------|--------------|-------|--------------|
|Speak|                |$Display|highlight_word(#offset)|

So, I think it *could* be done.

**Reply by sofia-bg** [reply]
It isn't a great idea for literacy but people are doing it for literacy and AAC so makes sense maybe might like to try it for testing out impact? I tried in CSS and all I was able to do was have entire message window changed to yellow after selecting, not individual words.

**Reply by gbishop** [reply]
Right. There is no way, currently, to address individual words. It could be added, I think.
